lib/iex/lib/iex/evaluator.ex

```diff
@@ -97,25 +97,33 @@ defmodule IEx.Evaluator do
     end
   end
 
-  defp do_eval(@break_trigger, state=%IEx.State{cache: ''}) do
+  defp do_eval(@break_trigger, state=%IEx.State{cache: ''}, _) do
     state
   end
 
-  defp do_eval(@break_trigger, state) do
+  defp do_eval(@break_trigger, state, _) do
     :elixir_errors.parse_error(state.counter, "iex", "incomplete expression", "")
   end
 
-  defp do_eval(latest_input, state) do
+  @history_key :__history_proc__
+
+  defp do_eval(latest_input, state, history) do
     code = state.cache ++ latest_input
     line = state.counter
-    handle_eval(Code.string_to_quoted(code, [line: line, file: "iex"]), code, line, state)
+
+    # Put the history pid into process dictionary so that IEx.Helpers can
+    # access it
+    Process.put(@history_key, history)
```

Comment:
Please correct me if I'm wrong, the purpose of "IEx history in a separate process" is to get rig of the all these pdict entries that evaluator process has out of the blue had.
Personally, I think that the most problematic part is solved - history itself is extracted, and probably it is tolerable to have history pid in the evaluator's pdict. At the same time I like the the idea of compete elimination of pdict usage for evaluator (mostly as an interesting challenge :smile:).